src/order/filter/pi.lean

```diff
@@ -0,0 +1,191 @@
+/-
+Copyright (c) 2021 Yury G. Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury G. Kudryashov, Alex Kontorovich
+-/
+import order.filter.bases
+
+/-!
+# (Co)product of a family of filters
+
+In this file we define two filters on `Π i, α i` and prove some basic properties of these filters.
+
+* `filter.pi (f : Π i, filter (α i))` to be the maximal filter on `Π i, α i` such that
+  `∀ i, filter.tendsto (function.eval i) (filter.pi f) (f i)`. It is defined as
+  `Π i, filter.comap (function.eval i) (f i)`. This is a generalization of `filter.prod` to indexed
+  products.
+
+* `filter.Coprod (f : Π i, filter (α i))`: a generalization of `filter.coprod`; it is the supremum
```

Comment:
Unrelated to this PR, but I guess this should be called `filter.copi`.

Reply:
I don't want to rename a filter in the same PR.